daffodil-core/src/main/scala/org/apache/daffodil/core/dpath/Expression.scala

```diff
@@ -1065,6 +1074,7 @@ sealed abstract class UpStepExpression(s: String, predArg: Option[PredicateExpre
     if (areAllArrays) {
       new CompiledDPath(UpMoveArray)
```

Comment:
Does the areAllArrays == true block need the predicate check too? This block means it the last up step with a targetType of array, so the expression could be something like `fn:count(..)`. If there is a predicate on this, e.g. `fn:count(..[1])` then I think that should have the same node-as-array an error since the the predicate would resolve to a single element of the array, and won't match the target type of array. I think? So it feels like regardless if the up-step is to an array or not we still never want to allow predicates?

Sidenote, UpMoveArray actually feels a little questionable based on these discussions--an up-step normally only goes to a single element so a special case where `..` goes to array feels odd? Seems like if you wanted the parents array you would have to do something like `fn:count(../../array)`. But maybe since we expect an array it's okay to assume the user meant the actual array and not the array element that up-step normally resolves to?

Reply:
Why allow "..[index]" at all? You could, in this case, do "../../array1_1[index]" I.e., use the array name. 
This is monomorphic in that the expression requires use of a specific array element name, but I'm not sure we lose much there. The only case I can think of where "../[index]" would be helpful is if you are creating some reusable element or group that wants to be polymorphic and work regardless of the name of the enclosing array.  

Polymorphism is good, but we have other reasons why we need "../a/*/b" i.e, wildcard stars, which is a more general way to get similar polymorphism. 

So I do not have a use case for this. "..[index]". It gives us some polymorphism without a need to extend the expression language with "*" wildcards. But it's only of theoretical concern unless there is some use case.

Reply:
Ok, so do we change `..` so it doesn't resolve to an array? Or should we leave it resolving to an array, but then disallow `..[index]` as Steve suggested?

Reply:
I think that makes sense. So `..` should never allow a predicate, whether it's an array or not.

But I think that also means that we need to remove `UpMoveArray`? That might be a more involved change and need some good testing, probably wants to be a separate bug/PR.